content/en/blog/2024/ambient-vs-cilium/index.md

```diff
@@ -0,0 +1,67 @@
+
+## Behind the Scenes: Why the Difference?
+
+The key to understanding these performance differences lies in the architecture and design of each tool.
+
+* **Cilium's Control Plane Conundrum:** Cilium runs a control plane instance on each node, leading to potential API server strain and configuration overhead as your cluster expands. This frequently caused our API Server to crash, followed by cilium becoming unready, and the entire cluster becoming unresponsive.
+* **Istio's Efficiency Edge:** Istio, with its centralized control plane and identity-based approach, streamlines configuration and reduces the burden on your API server, directing critical resources to processing and securing your traffic, rather than processing configuration. Istio takes further advantage of the resources not used in the control plane by running as many Envoy instances as a workload needs, while Cilium is limited to one shared Envoy instance per node.
+
+## Digging Deeper
+
+While the objective of this project is to compare Istio and Cilium scalability, several constraints make a direct comparison difficult.
+
+### Layer 4 Isn’t always Layer 4
+
+While Istio and Cilium both offer L4 policy enforcement, their APIs and implementation differ substantially. Cilium implements Kubernetes NetworkPolicy, which uses labels and namespaces to block or allow access to and from IP Addresses. Istio offers the AuthorizationPolicy API, and makes allow and deny decisions based on the TLS identity used to sign each request. Most defense-in-depth strategies will need to make use of both NetworkPolicy and TLS-based policy for comprehensive security.
+
+### Not all Encryption is Created Equal
+
+One challenge when comparing Istio and Cilium is the difference between how these projects approach encryption. While Cilium offers IPsec for FIPS-compatible encryption, most other Cilium features such as L7 Policy and Load Balancing are incompatible with IPsec. Cilium has much better feature compatibility when using Wireguard encryption, but Wireguard cannot be used in FIPS-compliant environments. Istio, on the other-hand, uses FIPS-compliant mTLS by default, which has the added benefit of complying with Zero-Trust principles.
+
+### Hidden Costs
+
+While Istio operates entirely in user space, Cilium’s L4 dataplane runs in the Linux Kernel using eBPF. Prometheus metrics for resource consumption only measure user space resources, meaning that all Kernel resources used by Cilium are not accounted for in this test.
+
+## Recommendations: Choosing the Right Tool for the Job
+
+So, what's the verdict? Well, it depends on your specific needs and priorities. If low latency is your absolute top priority and you have a smaller cluster, Cilium might still be a viable option. However, for larger clusters and a focus on stability, scalability, and advanced features, Istio Ambient Mode, along with an alternate NetworkPolicy implementation is the way to go.
```

Comment:
I am worried about this recommendation of low latency -> Cilium.

1. We are comparing Cilium L7 vs Istio L7 _only_ in this test. We have not proven any claims that any setup outside of this scope is faster or slower.
2. If we talk all about "Cilium L7", then say "use Cilium", the assumption is "Use Cilium [L7 mode]", which is not what the intended message here is (AFAIK)
3. I disagree with the assertion that Cilium L4 Wireguard is lower latency than Istio L4 TLS based on past testing.
4. Even if we are comparing apples-to-oranges of Cilium unencrypted to Ztunnel TLS, I still think the recommendation is wrong. Its like if a grilled hotdog has a 0.00000002% chance of causing health issues while a raw hotdog has a 0.00000001% chance --- the raw one may be 2x better, but _who cares_. Similar, IMO, applies to the latency overheads we are seeing with Ztunnel TLS. Adding 0.05ms to p90s is simply irrelevant to 99.99% of users

Reply:
I also don't know that we _need_ to recommend you use a networkpolicy on top of ambient...?

Reply:
I have substantially reworked this paragraph, which may address some of your concerns.  Without netpol, Ambient has no egress controls, which seem important from a defense standpoint...

Reply:
(1) still remains but it's a bit pedantic so probably fine

I'd note that with the new framing basically any CNI/kube-proxy/etc fits as well as cilium (just a note to myself really, not expecting a change to the blog...)